core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -964,6 +964,7 @@ public Playable getNextInQueue(Playable currentMedia) {
 
         @Override
         public void onPlaybackEnded(MediaType mediaType, boolean stopPlaying) {
+            PlaybackPreferences.clearCurrentlyPlayingTemporaryPlaybackSpeed();
```

Comment:
Good catch! Could you please move this to `PlaybackService.this.onPlaybackEnded`? I think it's more clean if everything is in one single place.